src/compiler/checker.ts

```diff
@@ -15059,7 +15066,10 @@ namespace ts {
         }
 
         function getTypeOfDestructuredProperty(type: Type, name: PropertyName) {
-            const text = getTextOfPropertyName(name);
+            const text = !isComputedNonLiteralName(name) ? getTextOfPropertyName(name) :

```

Comment:
Looks like there are now 2 positive uses of isComputedNonLiteralName to 4 negative uses. Since the name is already clunky, I think an inverted name like "isNormalName" or something is better. (We might need a new term for this kind of name.)

Reply:
It's unfortunate that getTextOfPropertyName can't delegate to getLateBoundName due to the variety of uses (at least in this PR). Maybe there is some way to unify them.

Reply:
The big question is whether we always want to get the late bound name in these cases. One option we might have is to stop using `getTextOfPropertyName` altogether and instead use `getLiteralTypeFromPropertyName` and pass that through `getLateBoundNameFromType` (which would work for non-late-bound names as well, and could possibly be renamed).